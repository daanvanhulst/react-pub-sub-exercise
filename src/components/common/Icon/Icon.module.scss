.o-icon {
  @apply flex-shrink-0;
}
