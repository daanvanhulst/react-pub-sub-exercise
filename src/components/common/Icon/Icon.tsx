import { icons } from 'lucide-react';
import styles from './Icon.module.scss';
import { cn } from '../../../helpers/cn';

export type IconNames = keyof typeof icons;

type IconProps = {
  name: IconNames;
  size?: 16 | 20 | 24 | 32 | 40 | 48 | 64 | 128;
  className?: string;
} & React.ComponentPropsWithRef<'svg'>;

export const Icon: React.FC<IconProps> = ({
  name,
  size = 24,
  className,
  ...restProps
}) => {
  const LucideIcon = icons[name];

  return (
    <LucideIcon
      {...restProps}
      size={size}
      className={cn(className, styles['o-icon'])}
    />
  );
};
