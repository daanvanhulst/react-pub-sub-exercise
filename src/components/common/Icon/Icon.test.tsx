import { render } from '@testing-library/react';
import { Icon } from 'src/components/common/Icon/Icon';
import { describe, expect, it } from 'vitest';

describe('Icon', () => {
  it('Renders correct one', () => {
    const test = render(<Icon name="X" />);
    expect(test).toMatchSnapshot();
  });

  it('Solid is rendering', () => {
    const test = render(<Icon name="X" />);
    expect(test).toMatchSnapshot();
  });
});
