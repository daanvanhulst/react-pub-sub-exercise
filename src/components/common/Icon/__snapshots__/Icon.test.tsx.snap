// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > Renders correct one 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="lucide lucide-x _o-icon_c087bc"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-x _o-icon_c087bc"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6 6 18"
      />
      <path
        d="m6 6 12 12"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon > Solid is rendering 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="lucide lucide-x _o-icon_c087bc"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-x _o-icon_c087bc"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6 6 18"
      />
      <path
        d="m6 6 12 12"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
